pkg/pillar/zedcloud/authen.go

```diff
@@ -514,10 +516,25 @@ func VerifySigningCertChain(log *base.LogObject, certs []*zcert.ZCert) ([]byte,
 			}
 		}
 	}
-	log.Tracef("VerifySigningCertChain: success\n")
 	return sigCertBytes, nil
 }
 
+type certStatus uint8
+
+const (
+	allCertsAreValid certStatus = iota
+	aCertIsInfo
+	aCertIsWarn
+	aCertIsINVALID
+)
+
+var (
+	lastCertTimeCheck time.Time
+	CurCertStatus     = allCertsAreValid
+	certInfoThreshold = 30
+	certWarnThreshold = 15
+)
+
 func verifySignature(log *base.LogObject, certByte []byte, interm *x509.CertPool) error {
```

Comment:
Note that this function does not get called until a new certificate chain is downloaded from the controller. We want a periodic verification that the chain is still valid, which means having zedagent run a timer to verify the chain.

Reply:
I'll take a look, but I thought it was executed with roughly the same frequency  of where I originally had the check code.  I obviously need to take a better look.